packages/plugin-connector/src/data-sources/build-sources.js

```diff
@@ -105,10 +109,13 @@ export const buildSources = async (
 			source.sourceDirectory
 		);
 		if (connectionValid !== true) {
+			logQueryEvent('db-connection-error', source.type, source.name);
 			throw new Error(
 				chalk.red(`[!] ${chalk.bold(source.name)} failed to connect; ${connectionValid.reason}`)
 			);
 		}
+		//TODO evidence-1344 and db-connection-error didn't have an equivalent for event in legacy
```

Comment:
@hughess in legacy we didn't emit an event when there was a connection error during query runs.  It was only emitted when the connection test failed.
Here I am doing it in both places.  Let me know if this not useful.